File: solutions/typescript/raindrops/1/raindrops.ts
const factors = [[3, "Pling"], [5, "Plang"], [7,  "Plong"]] as const

export function convert(n: number): string {
  const words = factors.reduce(
    (acc, [factor, word]) => n % factor === 0 ? acc + word : acc,
    ""
  )
  return words || n.toString()
}

File: solutions/scala/raindrops/1/src/main/scala/Raindrops.scala
object Raindrops {

  private final val factors = Map(3 -> "Pling", 5 -> "Plang", 7 -> "Plong")

  def convert(n: Int, factors: Map[Int, String] = factors): String = {
    val words = factors.collect { case (factor, word) if n % factor == 0 => word }
    if (words.nonEmpty) words.mkString else n.toString
  }

}

